compiler/optimizer/RematTools.cpp

```diff
@@ -316,3 +323,25 @@ void RematTools::walkTreeTopsCalculatingRematFailureAlternatives(TR::Compilation
       start = start->getNextTreeTop();
       }
    }
+
+/*
+ * This method is used to determine if indirect loads are safe for the nodes between
+ * the start inclusive and the end exclusive. It scans for indirect stores and
+ * determines the loads to be safe if none are found. The result can be passed to
+ * gatherNodesToCheck to allow remat of stores dependent on indirect loads.
+ */
+bool RematTools::checkForSafeIndirectLoads(TR::Compilation *comp, TR::TreeTop *start,
```

Comment:
Why are indirect stores sufficient to look for ? e.g. calls can also modify something that could be loaded indirectly right ? Perhaps one way would be to consult aliasing in this routine to assess which calls (and which stores, come to think of it) are worth panicking about.